templates/TestData.cpp.ejs

```diff
@@ -9,6 +9,69 @@
 #include <playfab/PlayFabAuthenticationAuthApi.h>
 #include <playfab/PlayFabClientDataModels.h>
 
+uint32_t g_testIndex = 1;
+
+<%
+function debugLogProp(varName, propName, propCoutName, property, prefix, padding) 
+{
+    var propType = getPublicPropertyType(property, prefix);
+    var propTypeClassName = getPublicPropertyTypeClassName(property, prefix);
+    if (propType == "PlayFabJsonObject") {%>
+<%- padding %>if( <%- varName %>-><%- propName %>.stringValue ) { std::cout << "  <%- propCoutName %> " << <%- varName %>-><%- propName %>.stringValue << std::endl; } else { std::cout << "  <%- propName %> = nullptr" << std::endl; }<%} else 
+if (propType == "PlayFabClientRegion" || propType == "PlayFabClientAdActivity") {%>
+<%- padding %>std::cout << "  <%- propCoutName %> " << (int)<%- varName %>-><%- propName %> << std::endl;<%} else 
+if (propType == "bool const*" || propType == "const char*") {%>
+<%- padding %>if( <%- varName %>-><%- propName %> ) { std::cout << "  <%- propCoutName %> " << <%- varName %>-><%- propName %> << std::endl; } else { std::cout << "  <%- propCoutName %> = nullptr" << std::endl; }<%} else {%>
+<%- padding %>std::cout << "  <%- propCoutName %> " << <%- varName %>-><%- propName %> << std::endl;<%} %> // Class: <%-propType%> <%
+}
+
+function debugLogCollection(varName, propName, property, prefix) {
+        var propType = getPublicPropertyTypeClassName(property, prefix);
+%>    
+    std::cout << "  <%- propName %>Count " << <%- varName %>-><%- propName %>Count << std::endl;
```

Comment:
>std::cout [](http://example.com/codeflow?start=4&length=9)

Can you put all this logging into a separate method rather than writing to cout directly?  That way we can easily redirect or disable the output later.

Reply:
Ya i can move to stringstream in next PR and then call some log method that does std out